components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/SessionDataPublisherImpl.java

```diff
@@ -0,0 +1,347 @@
+
+    }
+
+    @Override
+    public void doPublishAuthenticationStepFailure(AuthenticationData authenticationData) {
+
+    }
+
+    @Override
+    public void doPublishAuthenticationSuccess(AuthenticationData authenticationData) {
+
+    }
+
+    @Override
+    public void doPublishAuthenticationFailure(AuthenticationData authenticationData) {
+
+    }
+
+    @Override
+    public void doPublishSessionCreation(SessionData sessionData) {
+
+    }
+
+    @Override
+    public void doPublishSessionUpdate(SessionData sessionData) {
+
+    }
+
+    @Override
+    public void doPublishSessionTermination(SessionData sessionData) {
+
+    }
+
+    @Override
+    public String getName() {
+        return AuthPublisherConstants.DAS_SESSION_PUBLISHER_NAME;
+    }
+
+    @Override
+    public boolean isEnabled(MessageContext messageContext) {
+        return true;
+    }
+
+    /**
+     * Overridden method which implements the access token revocation
+     * @param request termination request
+     * @param context termination context
+     * @param sessionContext termination sessionContext
+     * @param params termination params
+     */
+    @Override
+    public void publishSessionTermination(HttpServletRequest request, AuthenticationContext context,
+            SessionContext sessionContext, Map<String, Object> params) {
+
+        OAuthConsumerAppDTO[] appDTOs = new OAuthConsumerAppDTO[0];
+        AuthenticatedUser authenticatedUser = (AuthenticatedUser) params.get("user");
+        String username = authenticatedUser.getUserName();
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        String userStoreDomain = authenticatedUser.getUserStoreDomain();
+        AuthenticatedUser federatedUser;
+
+        if (authenticatedUser.isFederatedUser()) {
+            try {
+                federatedUser = buildAuthenticatedUser(authenticatedUser);
+                authenticatedUser = federatedUser;
+
+            } catch (IdentityOAuth2Exception e) {
+                log.error("Error thrown while building authenticated user in logout flow for user " + authenticatedUser
+                        .getUserName() + " : " + e);
+            }
+        }
+        try {
+            appDTOs = getAppsAuthorizedByUser(authenticatedUser);
+            if (appDTOs.length > 0) {
+                if (log.isDebugEnabled()) {
+                    log.debug("The user: " + authenticatedUser.getUserName() + " has " + appDTOs.length + " "
+                            + "OAuth apps");
+                }
+            }
+        } catch (IdentityOAuthAdminException e) {
+            log.error("Error while retrieving applications authorized for the user " + authenticatedUser.getUserName()
+                    + " : " + e);
+        }
+
+        for (OAuthConsumerAppDTO appDTO : appDTOs) {
+            if (appDTO.getApplicationName().equalsIgnoreCase("admin_store") || appDTO.getApplicationName()
```

Comment:
Use of `StringUtils.equalIgnoreCase ` would be null safe.

Reply:
Changed the implementation by using StringUtils.equalIgnoreCase() method